classes/tables/entries_table.php

```diff
@@ -20,10 +20,34 @@
 
 require_once($CFG->libdir.'/tablelib.php');
 
+use moodle_url;
 use table_sql;
 
 class entries_table extends table_sql {
 
+    /**
+     * Format the column for the entry name.
+     *
+     * At the moment, this uses html_writer methods to add an update and delete button.
+     *
+     * @param $record
+     * @return string
+     * @throws \coding_exception
+     * @throws \moodle_exception
+     */
+    public function col_name($record) {
+        $cmid = required_param('id', PARAM_INT);
```

Comment:
it would be better if this cmid was passed into the table objects constructor and the required_param logic moved outside to the page that creates the object.